src/main/java/com/hubspot/httpql/impl/UriParamParser.java

```diff
@@ -158,4 +149,9 @@ public UriParamParser build() {
     }
   }
 
+  private static Optional<String> getFirst(Map<String, List<String>> map, String key) {
+    return Optional.ofNullable(map.get(key))
+        .filter(list -> !list.isEmpty())
+        .map(list -> list.get(0));
```

Comment:
Should this method handle the `orElse(null)` and return `String` instead of `Optional<String>` since it seems like all callers are doing that?

Reply:
👍  sure